aea/aea_builder.py

```diff
@@ -429,7 +431,7 @@ def set_decision_maker_handler(
 
         :return: self
         """
-        dotted_path, class_name = decision_maker_handler_dotted_path.split(":")
+        dotted_path, class_name = decision_maker_handler_dotted_path.split("::")
```

Comment:
We should create a constant for `::` and reuse throughout, The previous bug was precisely a result of us having these magic values